Mathlib/Analysis/Meromorphic/Divisor/MeromorphicFunction.lean

```diff
@@ -0,0 +1,124 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+
+import Mathlib.Analysis.Meromorphic.Order
+import Mathlib.Analysis.Meromorphic.Divisor.Basic
+
+/-!
+# The Divisor of a Meromorphic Function
+
+This file defines the divisor of a meromorphic function and proves the most
+basic lemmas about those divisors.
+
+## TODO
+
+- Remove the assumption `CompleteSpace E`.
+- Compatibility with restriction of divisors/functions
+- Non-negativity of the divisor for an analytic function
+- Behavior under addition of functions
+- Congruence lemmas for `codiscreteWithin`
+-/
+
+open Classical
+
+variable
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {U : Set 𝕜}
+  {z : 𝕜}
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E] [CompleteSpace E]
+
+namespace MeromorphicOn
+
+/-!
+## Definition of the Divisor
+-/
+
+/-- The divisor of a meromorphic function `f`, mapping a point `z` to the order
+  of `f` at `z`, and to zero if the order is infinite. -/
+noncomputable def divisor (f : 𝕜 → E) (hf : MeromorphicOn f U) :
```

Comment:
This definition is now `divisor`, but all the lemma names still mention `divisorOn`. I think they should become `divisor_*`.

Reply:
I guess an explicative is in order, but this might lock me out of GitHub. I am on the train now. Will fix as soon as I arrive.

Reply:
Fixed.